src/extension/sldService/src/main/java/org/geoserver/sldservice/rest/CapabilitiesController.java

```diff
@@ -0,0 +1,90 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.sldservice.rest;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import java.util.List;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.rest.RestBaseController;
+import org.geoserver.rest.converters.XStreamMessageConverter;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** CapabilitiesController. */
+@RestController
+@ControllerAdvice
+@RequestMapping(path = RestBaseController.ROOT_PATH + "/sldservice")
+public class CapabilitiesController extends BaseSLDServiceController {
+
+    public CapabilitiesController(Catalog catalog) {
+        super(catalog);
+    }
+
+    @GetMapping(
+        path = "/capabilities",
+        produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+    )
+    public Object classificationCapabilities() {
+        return wrapObject(new SldServiceCapabilities(), SldServiceCapabilities.class);
+    }
+
+    @Override
+    public void configurePersister(XStreamPersister persister, XStreamMessageConverter converter) {
+        XStream xstream = persister.getXStream();
+        xstream.alias("capabilities", SldServiceCapabilities.class);
+        xstream.registerConverter(new CapabilitiesConverter());
```

Comment:
It's a bit odd to see a custom converter being used, normally one would just let XStream handle the writing, aliasing classes and fields as necessary to get nicer names in output. It's not wrong though. Just wondering why you went down this road?

Reply:
thanks for your review @aaime. I tried by simply aliasing on fields but I wasn't able to get rid of @class attributeName. Also to have the array with methods named as "classifications" I would have needed to aliasing on type String, but this would have likely caused some issues if in the future more attribute would have been added to the SldServiceCapabilities class. I'm putting here the result obtained without the custom converter for clarity:
{
    "capabilities": {
        "vectorClassifications": {
            "@class": "java.util.Arrays$ArrayList",
            "string": [
                "quantile",
                "jenks",
                "equalArea",
                "equalInterval",
                "uniqueInterval",
                "standardDeviation"
            ]
        },
        "rasterClassifications": {
            "@class": "java.util.Arrays$ArrayList",
            "string": [
                "quantile",
                "jenks",
                "equalArea",
                "equalInterval",
                "uniqueInterval"
            ]
        }
    }
}